File: MasterDirectory/MasterDirectory.Web/Modules/ServerTypes/Merchandising/CatalogosMerchandisingForm.ts
﻿import { LookupEditor, IntegerEditor, StringEditor, BooleanEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface CatalogosMerchandisingForm {
    IdtipoCatalogo: LookupEditor;
    IdClave: IntegerEditor;
    Descripcion: StringEditor;
    Activo: BooleanEditor;
}

export class CatalogosMerchandisingForm extends PrefixedContext {
    static formKey = 'Merchandising.CatalogosMerchandising';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!CatalogosMerchandisingForm.init)  {
            CatalogosMerchandisingForm.init = true;

            var w0 = LookupEditor;
            var w1 = IntegerEditor;
            var w2 = StringEditor;
            var w3 = BooleanEditor;

            initFormType(CatalogosMerchandisingForm, [
                'IdtipoCatalogo', w0,
                'IdClave', w1,
                'Descripcion', w2,
                'Activo', w3
            ]);
        }
    }
}

File: MasterDirectory/MasterDirectory.Web/Modules/Merchandising/CatalogosMerchandising/CatalogosMerchandisingForm.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;
using System.ComponentModel.DataAnnotations;

namespace MasterDirectory.Merchandising.Forms;

[FormScript("Merchandising.CatalogosMerchandising")]
[BasedOnRow(typeof(CatalogosMerchandisingRow), CheckNames = true)]
public class CatalogosMerchandisingForm
{
    [Editable(false)]
    public int IdtipoCatalogo { get; set; }
    [Editable(false)]
    public int IdClave { get; set; }
    public string Descripcion { get; set; }
    [BooleanEditor]
    public int Activo { get; set; }
    //public DateTime DtRegistro { get; set; }
}